backend/BalanceBuddy/src/main/java/com/example/balancebuddy/controllers/GoalController.java

```diff
@@ -83,6 +105,7 @@ public ResponseEntity<Goal> addHabitToGoal(@PathVariable int id, @RequestBody Ad
     @PostMapping("/{id}/updateProgress")
     public ResponseEntity<Void> updateProgress(@PathVariable int id, @RequestBody ProgressUpdateDTO progressUpdateDTO) {
         goalService.updateProgress(id, progressUpdateDTO.getHabitName(), progressUpdateDTO.getProgressValue());
-        return new ResponseEntity<>(HttpStatus.OK);
+        //return new ResponseEntity<>(HttpStatus.OK);
+        return ResponseEntity.noContent().build();

```

Comment:
I think you should return "OK" status to indicate that the operation has succeded. Usually, the "NO_CONTENT" status is returned when you delete an entity.